### PR TITLE
Split reason parser and printer into sub-package

### PR DESCRIPTION
**Pull request by tekknolagi**
- [x] Split reason parser/printer off
- [ ] ~Split command-line tools~
  - [ ] ~Refmt~
  - [ ] ~Rebuild/Reasonbuild~
  - [ ] ~Rtop~
  - [ ] ~Reason utils~
- [ ] ~Update bspack situation (cc @chenglou)~ I broke it elsewhere
- [x] Edit README in `src` to reflect changes

### CODE REVIEW

**Comment by tekknolagi** [comment]
Also this won't install on CI until reason-parser is published, I guess.

**Comment by tekknolagi** [comment]
reason-parser is published on opam at 1.12.0 and I am writing some scripts to aid in deploying sub-packages.

**Comment by tekknolagi** [comment]
Since bspack is broken by my other PR adding ocaml-migrate parsetree, I can't fix it with this diff.

cc @yunxing for review

**Comment by chenglou** [comment]
Update for bspack: got some progress. This diff itself looks good.

File: scripts/make-subpkg.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
```
Comment (yunxing) [inline scripts/make-subpkg.sh]: Who calls into this script?

**Comment by tekknolagi** [comment]
Manual. May be rendered useless by opam-publish. TBD.

File: scripts/make-subpkg.sh (hunk 1 above)
Comment (tekknolagi) [inline scripts/make-subpkg.sh]: Oops I left this as another comment. "Manual. May be rendered useless by opam-publish. TBD."

Comment (yunxing) [inline scripts/make-subpkg.sh]: Still trying to understand the workflow. How do you currently upload the tar file?

**Comment by tekknolagi** [comment]
@yunxing updated the release script to use GitHub API. Now will also attempt to create and release opam packages too.

**Pull request merged**